core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableJoin.java

```diff
@@ -64,27 +77,20 @@ protected EnumerableJoin(
         traits,
         left,
         right,
-        condition,
+        equiCondition,
         leftKeys,
         rightKeys,
         variablesSet,
         joinType);
-  }
-
-  @Deprecated // to be removed before 2.0
-  protected EnumerableJoin(RelOptCluster cluster, RelTraitSet traits,
-      RelNode left, RelNode right, RexNode condition, ImmutableIntList leftKeys,
-      ImmutableIntList rightKeys, JoinRelType joinType,
-      Set<String> variablesStopped) throws InvalidRelException {
-    this(cluster, traits, left, right, condition, leftKeys, rightKeys,
-        CorrelationId.setOf(variablesStopped), joinType);
+    this.remainCondition = remainCondition;
   }
 
   /** Creates an EnumerableJoin. */
   public static EnumerableJoin create(
       RelNode left,
       RelNode right,
-      RexNode condition,
+      RexNode equiCondition,
+      RexNode remainCondition,
```

Comment:
To avoid changing this public method, and considering that `remainCondition` can be null, maybe it would be safer to keep this method declaration as it is, and create a new one with the new argument `RexNode remainCondition` (and of course the former would call the latter passing `null` as `remainCondition`).

Reply:
Agree with you.

Reply:
@rubenada ,I reverted the old constructor method and create method in latest commit.

Reply:
Thanks @LaiZhou , but I was talking about the factory `public static EnumerableJoin create` method, sorry if I was not clear.

Reply:
@rubenada ,sorry for my mistake. Now I make the old create method  to be delegated to the new create method .

Reply:
No problem, thanks for the change. One last (small) thing, maybe we should move the new constructor together with the other ones (instead of in between the several `create` methods)

Reply:
Will do it later.

Reply:
done.